### PR TITLE
Give the IS a bunch more 3pid invite context

### PR DESCRIPTION
**Pull request by illicitonion**
This allows it to form richer emails

### CODE REVIEW

File: synapse/handlers/room.py [hunk 1]
```diff
@@ -704,13 +704,50 @@ def _make_and_store_3pid_invite(
...
                 address,
                 room_id,
-                user.to_string()
+                canonical_room_alias,
+                room_avatar_url,
+                room_join_rules,
+                room_name,
+                user.to_string(),
+                inviter_display_name,
+                inviter_avatar_url
```
Comment (Kegsay) [inline synapse/handlers/room.py]: Please can we kwarg this, this looks horribly error-prone.

Comment (illicitonion) [inline synapse/handlers/room.py]: Done

File: synapse/handlers/room.py [hunk 2]
```diff
@@ -704,13 +704,56 @@ def _make_and_store_3pid_invite(
...
+
+        canonical_room_alias = ""
+        canonical_alias_event = room_state.get((EventTypes.CanonicalAlias, ""))
+        if canonical_alias_event and canonical_alias_event.content["alias"]:
+            canonical_room_alias = canonical_alias_event.content["alias"]
+
+        room_name = ""
+        room_name_event = room_state.get((EventTypes.Name, ""))
+        if room_name_event and room_name_event.content["name"]:
+            room_name = room_name_event.content["name"]
```
Comment (NegativeMjark) [inline synapse/handlers/room.py]: This probably should check that the key "name" is present.

     room_name_event.content.get("name", "")

File: synapse/handlers/room.py [hunk 3]
```diff
@@ -704,13 +704,56 @@ def _make_and_store_3pid_invite(
...
+
+        room_name = ""
+        room_name_event = room_state.get((EventTypes.Name, ""))
+        if room_name_event and room_name_event.content.get("name", ""):
+            room_name = room_name_event.content["name"]
+
+        room_join_rules = ""
+        join_rules_event = room_state.get((EventTypes.JoinRules, ""))
+        if join_rules_event and join_rules_event.content.get("join_rule", ""):
+            room_join_rules = join_rules_event.content["join_rule"]
```
Comment (NegativeMjark) [inline synapse/handlers/room.py]: You could do:

    room_join_rules = join_rules_event.content.get("join_rule", "")

**Comment by NegativeMjark** [comment]
LGTM

**Pull request merged**